dev_guide/expose_external_ip.adoc

```diff
@@ -0,0 +1,781 @@
+  externalIPNetworkCIDRs:
+  - <ip_address>/<cidr>
+----
++
+where `<ip_address>/<cidr>` is a range of IP addresses you plan to assign to the service(s) you want to access externally.
++
+For example:
++
+----
+networkConfig:
+  externalIPNetworkCIDRs:
+  - 192.168.120.0/24
+----
+
+. Restart the {product-title} master service to apply the changes.
++
+ifdef::openshift-enterprise[]
+----
+# systemctl restart atomic-openshift-master
+----
+endif::[]
+ifdef::openshift-origin[]
+----
+# systemctl restart origin-master
+----
+endif::[]
+
+[CAUTION]
+====
+You must make sure that the IP address pool you assign terminates at one or more nodes in your cluster. You can use the existing
+xref:../admin_guide/high_availability.adoc#configuring-ip-failover[`*oadm ipfailover*`] to ensure that the external IPs are highly available.
+====
+
+[[exposing-service]]
+=== Expose the Service to Create a Route
+
+This task is required for the xref:manually-assign-ips[service IP] and xref:automatically-assign-lb-service[load balancer] solutions.
+
+You must xref:../cli_reference/basic_cli_operations.html#expose[expose the service as a route] using the `oc expose` command.
+
+To expose the service:
+
+. Make sure you have xref:end-to-end-login[logged into]  {product-title} master node as a user with the cluster admin role.
+
+. Log into the xref:create-external-project-service[project] where the service you want to expose is located.
++
+----
+$ oc project project1
+----
+
+. Execute the following command to expose the route:
++
+----
+oc expose service <service-name>
+----
++
+For example:
++
+----
+oc expose service mysql-55-rhel7
+route "mysql-55-rhel7" exposed
+----
+
+. On the master, use a tool, such as cURL, to make sure you can reach the service using the cluster IP address for the service:
++
+----
+curl <pod-ip>:<port>
+----
++
+For example:
++
+----
+curl:172.30.131.89:3306
+----
++
+The examples in this section use a MySQL service, which requires a client application. If you receive a cURL response similar to the following, you are connecting with the service:
++
+----
+curl 172.30.131.89:3306
+5.7.16�Z(7uPs*u���T(YSNF6t.mysql_native_password��Got packets out of order
+----
+If you have a MySQL client, log in with the standard CLI command:
++
+----
+$ mysql -h 172.30.131.89 -u admin -p
+Enter password:
+Welcome to the MariaDB monitor.  Commands end with ; or \g.
+
+MySQL [(none)]>
+----
+
+[[expose-service-router]]
+== Using a Router
+
+Using a router is the most common way xref:index.adoc#getting-traffic-into-cluster-index[to allow external access to an {product-title} cluster].
+
+Work with your administrators to configure a router
+to accept external requests and proxy them based on the
+configured routes. This is limited to HTTP/HTTPS(SNI)/TLS(SNI), which
+covers web applications.
+
+The administrator can create a
+ifdef::openshift-enterprise,openshift-origin[]
+xref:../install_config/install/prerequisites.adoc#prereq-dns[wildcard DNS]
```

Comment:
This link appears to be broken, but it could just be something wonky with the ifdef in preview.